Sync v2-10-stable with v2-10-test to release 2.10.0rc1

utkarsharma2 opened the pull request:
Time for 2.10.0rc1!

utkarsharma2 commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,333 @@
+
+Decorator for Task Flow, to make it simple to apply whether or not to skip a Task. (#41116)
+"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+This feature adds a decorator to make it simple to skip a Task.
```
Here, the description doesn't add any more value than the title, I have just added it as a placeholder. Suggestions are welcomed. :)

phi-friday replied:
That's really all there is to it, so I'm not sure what else to add.
This uses `pre_exeuce()`, so if `pre_execute()` is still an experimental feature in `2.10.x`, it would be nice to note that.

jedcunningham replied:
My 2c, this can just be a new feature line. It's not noteworthy enough to warrant a significant entry.

kaxil replied:
Is that a decorator for Branching?

kaxil replied:
But otherwise agreed with Jed

ephraimbuddy replied:
I added a little bit more info to the title

utkarsharma2 commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,333 @@
+""""""""""""""""""""""""""""""""""""""""""""""
+Previously known as hybrid executors, this new feature allows Airflow to use multiple executors concurrently. DAGs, or even individual tasks, can be configured
+to use a specific executor that suits its needs best. A single DAG can contain tasks all using different executors. Please see the Airflow documentation for
+more details. Note: This feature is still experimental. See `documentation on Executor <https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/executor/index.html#executor>`_ for a more detailed description.
```
@jedcunningham I have linked the doc of the executor here since the doc we want to refer is not published yet. Please let me know if this needs to be changed.

jedcunningham replied:
See the other suggestion. We can deep link now since we know what it will be.

kaxil replied:
Yup, we can cross reference and deep link

kaxil replied:
https://www.sphinx-doc.org/en/master/usage/referencing.html#cross-referencing-arbitrary-locations should work too

jedcunningham replied:
We could! I think having the link work from github has tremendous value though. Idk about others, but I use githubs copy of the release notes almost exclusively (I could be biased 😄)

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,333 @@
+
+- Decorator for Task Flow, to make it simple to apply whether or not to skip a Task. (#41116)
+
+
```
Double 2.10s! This should be folded in below.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -436,6 +763,8 @@ Improvements
 - Remove dependency of ``Connexion`` from auth manager interface (#36209)
 - Refactor ExternalDagLink to not create ad hoc TaskInstances (#36135)
+- Allowing DateTimeSensorAsync, FileSensor and TimeSensorAsync to start execution from trigger during dynamic task mapping (#41182)
+- Call listener on_task_instance_failed() after ti state is changed (#41053)
```
Shouldn't these be in the 2.10 section?

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,333 @@
+- Improve Task TryHistory UI (#40357)
+- Improve UI error handling (#40350)
+- Remove double warning in CLI  when config value is deprecated (#40319)
+- AIP 64: Add TI try history to Task Instance Details, Logs, and Gantt chart (#40304)
```
```suggestion
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,333 @@
+- Refactor DAG.dataset_triggers into the timetable class (#39321)
+- Show TI history for log links, more ti details and mapped tasks (#40375)
+- add next_kwargs to StartTriggerArgs (#40376)
+- Improve Task TryHistory UI (#40357)
```
```suggestion
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,333 @@
+- Attempt to add ReactJSON view to rendered templates (#40639)
+- Speeding up regex match time for custom warnings (#40513)
+- Refactor DAG.dataset_triggers into the timetable class (#39321)
+- Show TI history for log links, more ti details and mapped tasks (#40375)
```
```suggestion
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,333 @@
+""""""""""""""""
+- Add ``filesystems`` and ``dataset-uris`` items to How to create your own (#40801)
+- Fix (TM) to (R) in Airflow repository (#40783)
+- docs: set otel_on to True in example airflow.cfg (#40712)
```
```suggestion
- Set ``otel_on`` to True in example airflow.cfg (#40712)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,333 @@
+- Add ``filesystems`` and ``dataset-uris`` items to How to create your own (#40801)
+- Fix (TM) to (R) in Airflow repository (#40783)
+- docs: set otel_on to True in example airflow.cfg (#40712)
+- Documentation: Warning for _AIRFLOW_PATCH_GEVENT  (#40677)
```
```suggestion
- Add warning for _AIRFLOW_PATCH_GEVENT (#40677)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,333 @@
+- docs: set otel_on to True in example airflow.cfg (#40712)
+- Documentation: Warning for _AIRFLOW_PATCH_GEVENT  (#40677)
+- Update multi-team diagram proposal after Airflow 3 discussions (#40671)
+- MSSQL-DOC-FIX (#40565)
```
```suggestion
- Add stronger warning that MSSQL is not supported and no longer functional (#40565)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,333 @@
+- Documentation: Warning for _AIRFLOW_PATCH_GEVENT  (#40677)
+- Update multi-team diagram proposal after Airflow 3 discussions (#40671)
+- MSSQL-DOC-FIX (#40565)
+- Update best-practices.rst (#40505)
```
```suggestion
```

My 2c, skip this one. It's basically a grammar fix.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,333 @@
+- Fix create a local virtualenv example (#40373)
+- Change the pre-commit installing command to use pipx instead of ``pyenv`` (#40266)
+- fix sub-heading in 07_local_virtualenv.rst (#39995)
+- Replace non-functional TOC directive with explanation where to look (#39901)
```
```suggestion
```

These aren't core.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,333 @@
+- Replace non-functional TOC directive with explanation where to look (#39901)
+- update k8s supported version in docs (#39878)
+- Add compatibility note for Listeners (#39544)
+- Fix typo sensitive masking words in docs (#39415)
```
```suggestion
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,333 @@
+- update k8s supported version in docs (#39878)
+- Add compatibility note for Listeners (#39544)
+- Fix typo sensitive masking words in docs (#39415)
+- Fix typo in templates-ref.rst (#38865)
```
```suggestion
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,333 @@
+- Add information about lack of security guarantees for docker compose (#41072)
+- Add links to example dags in use params section (#41031)
+- Change ``task_id`` from ``send_email`` to ``send_email_notification`` in ``taskflow.rst`` (#41060)
+- Add more detail on when dataset alias is resolved
```
Hmm, where is the PR number on this one?

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,333 @@
+""""""""""""""""""""""""""""""""""""""""""""""
+Previously known as hybrid executors, this new feature allows Airflow to use multiple executors concurrently. DAGs, or even individual tasks, can be configured
+to use a specific executor that suits its needs best. A single DAG can contain tasks all using different executors. Please see the Airflow documentation for
+more details. Note: This feature is still experimental. See `documentation on Executor <https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/executor/index.html#executor>`_ for a more detailed description.
```
```suggestion
more details. Note: This feature is still experimental. See `documentation on Executor <https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/executor/index.html#using-multiple-executors-concurrently>`_ for a more detailed description.
```

We know what it will be, however, so let's do the deep link.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,333 @@
+- Use walrus to remove one line of code (#39906)
+- Refactor BashOperator and Bash decorator for consistency and simplicity (#39871)
+- Much simpler way to run provider compatibility tests (#39883)
+- Run tests for Providers also for Airflow 2.8 (#39606)
```
```suggestion
```

These don't need to be on user facing release notes.

kaxil commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,333 @@
+This feature adds a decorator to make it simple to skip a Task.
+
+
+New Features
```
Could we add a note about Telemetry collection please

ephraimbuddy commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,333 @@
+- Dag test without sensor (#40010)
+- Handle db isolation for mapped operators and task groups (#39259)
+- feature: callable for template_fields (#37028)
+- Home: filter running/failed and active/paused dags (#39701)
```
```suggestion
- Filter running/failed and active/paused dags  at the home page(#39701)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Use sentinel to mark dag as removed on re-serialization (#39825)
+- Add parameter for the last number of queries to the DB in DAG file processing stats (#40323)
+- Add prototype version dark mode for Airflow UI (#39355)
+- Move import to make PythonOperator working on Windows (#40424)
```
bugfix?

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Add parameter for the last number of queries to the DB in DAG file processing stats (#40323)
+- Add prototype version dark mode for Airflow UI (#39355)
+- Move import to make PythonOperator working on Windows (#40424)
+- Dag test without sensor (#40010)
```
```suggestion
- Add ability to mark some tasks as successful in ``dag test``  (#40010)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Add prototype version dark mode for Airflow UI (#39355)
+- Move import to make PythonOperator working on Windows (#40424)
+- Dag test without sensor (#40010)
+- Handle db isolation for mapped operators and task groups (#39259)
```
```suggestion
```

This is covered by AIP-44.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Dag test without sensor (#40010)
+- Handle db isolation for mapped operators and task groups (#39259)
+- Allow use of callable for template_fields (#37028)
+- Filter running/failed and active/paused dags  at the home page(#39701)
```
```suggestion
- Filter running/failed and active/paused dags on the home page (#39701)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- UI changes for DAG Re-parsing feature (#39636)
+- Add Scarf based telemetry (#39510)
+- Add dag re-parsing request endpoint (#39138)
+- Task adoption for hybrid executors (#39531)
```
```suggestion
```

This is part of AIP-61.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Make audit log before/after filterable (#39120)
+- Consolidate grid collapse actions to a single full screen toggle (#39070)
+- Implement Metadata to emit runtime extra (#38650)
+- Add ``AirflowInternalRuntimeError`` for raise ``non catchable`` errors (#38778)
```
misc feels more appropriate for this one.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Add ``AirflowInternalRuntimeError`` for raise ``non catchable`` errors (#38778)
+- Add executor field to the DB and parameter to the operators (#38474)
+- Implement context accessor for DatasetEvent extra (#38481)
+- Add health endpoint to the RPC server (#38551)
```
```suggestion
```

AIP-44

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Add dag_stats rest api endpoint (#41017)
+- Add listeners for Dag import errors (#39739)
+- Allowing DateTimeSensorAsync, FileSensor and TimeSensorAsync to start execution from trigger during dynamic task mapping (#41182)
+- Call listener on_task_instance_failed() after ti state is changed (#41053)
```
bugfix

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Determine needs_expansion at time of serialization (#39604)
+- Add indexes on dag_id column in referencing tables to speed up deletion of dag records (#39638)
+- Add task failed dependencies to details page (#38449)
+- Support failing tasks stuck in queued for hybrid executors (#39624)
```
```suggestion
```

AIP-61

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Use ``declarative_base`` from ``sqlalchemy.orm`` instead of ``sqlalchemy.ext.declarative`` (#39134)
+- Add example DAG to demonstrate emitting approaches (#38821)
+- Give ``on_task_instance_failed`` access to the error that caused the failure (#38155)
+- Make _get_ti compatible with RPC (#38570)
```
```suggestion
```

AIP-44

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Give ``on_task_instance_failed`` access to the error that caused the failure (#38155)
+- Make _get_ti compatible with RPC (#38570)
+- Simplify dataset serialization (#38694)
+- recovery message (#34457)
```
```suggestion
- Add heartbeat recovery message to jobs (#34457)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Remove select_column option in TaskInstance.get_task_instance (#38571)
+- Don't create session in get_dag if not reading dags from database (#38553)
+- Update render filename to use internal API (#38558)
+- Ensure orm models loaded when using RPC API (#38568)
```
```suggestion
```

AIP-44

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Ensure orm models loaded when using RPC API (#38568)
+- Add a migration script for encrypted trigger kwargs (#38358)
+- Implement render_templates on TaskInstancePydantic (#38559)
+- Don't dispose pools when using internal api (#38552)
```
```suggestion
```

AIP-44

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Don't dispose pools when using internal api (#38552)
+- Handle optional session in _refresh_from_db (#38572)
+- Ensure that task is set properly when ti is TaskInstancePydantic (#38547)
+- Implement _run_execute_callback on TaskInstancePydantic (#38560)
```
```suggestion
```

AIP-44

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Bugfix for get_parsing_context() when ran with LocalExecutor (#40738)
+- Validating provider documentation urls before displaying in views (#40933)
+- Fix dataset_with_extra_from_classic_operator example DAG (#40747)
+- fix: never_fail in sensor (#40915)
```
```suggestion
- Add ``never_fail`` in BaseSensor (#40915)
```

This should be an improvement or new feature. But I also think it's a breaking change :(

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- fix: never_fail in sensor (#40915)
+- Fix tasks API endpoint when DAG doesn't have ``start_date`` (#40878)
+- Prevent SubdagOperator to fail when Database Isolation is enabled (#40771)
+- Prevent NoneType Error at plugin loading with database isolation mode (#40770)
```
```suggestion
```

AIP-44

ephraimbuddy committed:
regenerate command hashes

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Only heartbeat if necessary in backfill loop (#39399)
+- Fix trigger kwarg encryption migration (#39246)
+- Use ``model_validate`` instead of ``parse_obj`` for de-serialize Pydantic V2 model (#38999)
+- Fix try_number handling when db isolation enabled (#38943)
```
```suggestion
```

AIP-44

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Fix decryption of trigger kwargs when downgrading. (#38743)
+- Fix duplicate "health" line in internal api openapi config (#38661)
+- Don't double-serialize params node in RPC calls (#38548)
+- Don't check migrations when internal API enabled (#38556)
```
```suggestion
```

AIP-44

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Fix wrong link in TriggeredDagRuns (#41166)
+- Pass MapIndex to LogLink component for external log systems (#41125)
+- Add NonCachingRotatingFileHandler for worker task (#41064)
+- Fix broken/renamed internal API method (#41070)
```
```suggestion
```

AIP-44

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Make HookLineageCollector group datasets by (#41034)
+- Enhance start_trigger_args serialization (#40993)
+- Refactor ``BaseSensorOperator`` introduce ``skip_policy`` parameter (#40924)
+- Check served logs logic (#41272)
```
```suggestion
- Fix viewing logs from triggerer when task is deferred (#41272)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Clean up the exception handler when run_as_user is the airflow user  (#41241)
+- Collapse docs when click and folded (#41214)
+- Update updated_at when saving to db as session.merge does not trigger on-update (#40782)
+- Query count statistics when parsing DAF file (#41149)
```
```suggestion
- Fix query count statistics when parsing DAG file (#41149)
```

ephraimbuddy committed:
Update version to 2.10.0

ephraimbuddy committed:
Update RELEASE_NOTES.rst

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,301 @@
+- Widen cheat sheet column to avoid wrapping commands (#38888)
+- Update hatchling to latest version (1.22.5) (#38780)
+- bump uv to 0.1.29 (#38758)
+- Don't actually check the db when using internal API (#38554)
```
```suggestion
```

AIP-44

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,6 +21,318 @@
+
+Doc Only Changes
+""""""""""""""""
+- Add ``filesystems`` and ``dataset-uris`` items to How to create your own (#40801)
```
```suggestion
- Add ``filesystems`` and ``dataset-uris`` to "how to create your own provider" page (#40801)
```

bbovenzi committed:
Fix Gantt Task Tries (#41342)

(cherry picked from commit cb80bda5eedb3a52fa786df81d5e7eb14caf31ff)

ephraimbuddy committed:
Revert "Send context using in venv operator (#41039)" (#41362)

This reverts commit da553935d248f22695124c40777d3ea29e04d57f.

(cherry picked from commit 0a06cc683a877d63305060766ae78f9ffe1fba0e)

jscheffl committed:
Fix tests/models/test_taskinstance.py for Database Isolation Tests (#41344)

(cherry picked from commit f811ac304e3d49b015ffb1e326863fe1cce14523)

jscheffl committed:
Fix TriggerDagRunOperator Tests for Database Isolation Tests (#41298)

* Attempt to fix TriggerDagRunOperator for Database Isolation Tests

* Finalize making tests run for triggerdagrunoperator in db isolation mode

* Adjust query count assert for adjustments to serialization

* Review feedback

(cherry picked from commit 6b810b89c3f63dd2d2cf107c568be40ba9da0ba2)

jscheffl committed:
Fix core tests from start to SkipMixin for Database Isolation Mode (#41369)

* Skip core tests from start to SkipMixin for Database Isolation Mode

* Skip core tests from start to SkipMixin for Database Isolation Mode, uups

* Skip core tests from start to SkipMixin for Database Isolation Mode, uups

(cherry picked from commit b87f9878ff801575b35a30220c0b2e285a2cf1b8)

ephraimbuddy committed:
Retain the function "resource_name_for_dag" for backwards compatibility (#41382)

When using older FAB providers on the new airflow, this function is called
in the old provider and is no longer available in the new airflow. This
PR brings this back to fix issue in main and v2-10-test branch where all
DAGs fail because of lack of this function

(cherry picked from commit 0576f557f1fccf059cd9f70ec9fd9a1fb555cfec)

bugraoz93 committed:
Fix Variable and KubernetesJobOperator Tests for Database Isolation Tests (#41370)

Fixing remaining Variable tests for db isolation mode, also fixing secret backend haven't called from EnvironmentVariablesBackend, Metastore and custom ones. This caused side effect to move the Variable.get() method to internal API

(cherry picked from commit c98d1a177be4b260d44df75f6e97f752bb381cce)

jscheffl committed:
Fix pytests for Core except Variable for DB Isolation Mode (#41375)

(cherry picked from commit 60cbea509707a4390ea9ea4239c737f0a44e8aea)

dirrao committed:
bump uv version to 0.2.34 (#41334)

(cherry picked from commit b4a92f802e0f6c1aac09953e4c490a149b5e7c12)

potiuk committed:
Skip docs publishing on non-main brnaches (#41385)

(cherry picked from commit 54c165cee03808bcf6a83708b5a58f2b205a3d9d)

potiuk committed:
Fix mypy checks for new azure libraries (#41386)

(cherry picked from commit 68a6a0583b5100c1f313d7bff3dc664205d9a1ad)

bugraoz93 committed:
Fix  tests/decorators/test_python.py for database isolation tests (#41387)

* Pass serialized parameter for dag_maker

* Serialisation of object is on __exit__ moving out the dag definition out of dag_maker context

(cherry picked from commit 278f3c4b6eaf823a0c6eb538a6f61d21e0f955d8)

Lee-W committed:
fix DagPriorityParsingRequest unique constraint error when dataset aliases are resolved into new datasets (#41398)

* fix(datasets/manager): fix DagPriorityParsingRequest unique constraint error when dataset aliases are resolved into new datasets

this happens when dynamic task mapping is used

* refactor(dataset/manager): reword debug log

Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

* refactor(dataset/manager): remove unnecessary logging

Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

---------

Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>
(cherry picked from commit bf64cb686be28f5e60e0b06f5fae790481c2efcd)
